rust: Add support of IEEE 802.1X authentication

cathay4t opened the pull request:
Integration test case enabled.

The changes to `tests/integration/nm/conftest.py` is moving its pytest 
fixture to it actual user `tests/integration/nm/checkpoint.py`. The 
functions used by that fixture is not nmstate public API, there is no plan
on support it in rust yet.

ffmancera commented on rust/src/lib/ieee8021x.rs:
```diff
@@ -0,0 +1,29 @@
+use crate::NetworkState;
+
+#[derive(Debug, Clone, PartialEq, Default, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct Ieee8021XConfig {
```
Please add the `non-exhaustive` as we do not want to include new structs without it.

ffmancera commented on rust/src/libnm_dbus/connection/ieee8021x.rs:
```diff
@@ -0,0 +1,123 @@
+const GLIB_FILE_PATH_PREFIX: &str = "file://";
+
+#[derive(Debug, Clone, PartialEq, Default, Deserialize)]
+#[serde(try_from = "DbusDictionary")]
+pub struct NmSetting8021X {
```
Please add the `non-exhaustive` as we do not want to include new structs without it.

ffmancera commented on tests/integration/nm/conftest.py:
```diff
@@ -16,27 +16,3 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with this program. If not, see <https://www.gnu.org/licenses/>.
 #
-import pytest
-
-from libnmstate.nm.plugin import NetworkManagerPlugin
-from libnmstate.nm.context import NmContext
-
-
-@pytest.fixture(scope="function")
-def nm_plugin():
-    plugin = NetworkManagerPlugin()
-    yield plugin
-    if plugin.checkpoint:
-        # Ignore failures as the checkpoint might already expired
-        try:
-            plugin.rollback_checkpoint()
-        except Exception:
-            pass
-    plugin.unload()
-
-
-@pytest.fixture(scope="function")
-def nm_context():
-    ctx = NmContext()
-    yield ctx
-    ctx.clean_up()
```
Why is this being moved? And if this is needed (I do not see why), I think you should remove the whole file as the remaining is only the header. Thanks!

cathay4t replied:
The changes to tests/integration/nm/conftest.py is moving its pytest
fixture to it actual user tests/integration/nm/checkpoint.py

`contest.py` removed.

cathay4t committed:
rust: Add support of IEEE 802.1X authentication

Integration test case enabled.

All the code in `tests/integration/nm/conftest.py` have moved its actual
user `tests/integration/nm/checkpoint.py`. The functions used by that
fixture is not nmstate public API, there is no plan on support it in
rust yet.

Signed-off-by: Gris Ge <fge@redhat.com>
